common/common.cpp

```diff
@@ -2076,6 +2077,17 @@ static bool llama_download_file(const std::string & url, const std::string & pat
             }
         }
 
+        // Create parent directories if not exist
+        const std::vector<std::string> path_parts = string_split(path_temporary, DIRECTORY_SEPARATOR);
```

Comment:
Can we reuse the existing `bool create_directory_with_parents(const std::string & path);`